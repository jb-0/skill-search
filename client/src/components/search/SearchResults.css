.search-results-container {
  color: white;
  font-size: 1.4rem;
}