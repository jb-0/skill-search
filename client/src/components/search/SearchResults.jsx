import React from 'react';
import './SearchResults.css'

function SearchResults(props) {

  return (
    <div className="search-results-container">
      <p>
        There are currently {props.jobs.noOfResults} "{props.searchTerms.join(' ')}" jobs in 
        {' '} {props.location}
      </p>
    </div>
  );
};

export default SearchResults;