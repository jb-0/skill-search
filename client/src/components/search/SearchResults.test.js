import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

import SearchResults from './SearchResults';

describe('SearchResults component', () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('props passed are rendered successfully and prefixed with "x"', () => {
    act(() => {
      render(
        <SearchResults
          searchTerms={['node', 'react']}
          location={'london'}
          jobs={{ noOfResults: 254, msg: 'results found' }}
        />,
        container
      );
    });

    expect(
      screen.getByText('There are currently 254 "node react" jobs in london')
    ).toBeInTheDocument();
  });

  it('renders correctly when passed props', () => {
    const tree = renderer
      .create(
        <SearchResults
          searchTerms={['node', 'react']}
          location={'london'}
          jobs={{ noOfResults: 254, msg: 'results found' }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
