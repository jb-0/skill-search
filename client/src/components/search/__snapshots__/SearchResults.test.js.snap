// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders correctly when passed props 1`] = `
<div
  className="search-results-container"
>
  <p>
    There are currently 
    254
     "
    node react
    " jobs in
     
     
    london
  </p>
</div>
`;
